File: src/Api/Model/DB/Municipio.cs
using Newtonsoft.Json;
using Realms;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CriadorBaseDados.Model.DB
{
    [JsonObject(MemberSerialization.OptIn)]
    public class Municipio : RealmObject
    {
        [JsonProperty]
        [PrimaryKey]
        public int ID { get; set; }
        [JsonProperty]
        public string Nome { get; set; }
        [Ignored]
        [JsonProperty]
        public string CodUF { get => UF.UF; }
        public Estado UF { get; set; }
        [Backlink(nameof(Empresa.Municipio))]
        public IQueryable<Empresa> Empresas { get;}
    }
}

File: src/Mobile/b2b/lib/model/core/municipio.dart
import 'empresa.dart';

class Municipio {
  final int id;
  final String nome;
  final String uf;
  final List<Empresa> empresas;

  Municipio(this.id, this.nome, this.uf, this.empresas);

  Municipio.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        nome = json['nome'],
        uf = json['uf'],
        empresas = json['empresas'] ?? List.empty(growable: true);

  Map<String, dynamic> toMap() => <String, dynamic>{
        'id': id,
        'nome': nome,
        'uf': uf,
        //'empresas': empresas,
      };
}
